src/boot/efifirmware.c

```diff
@@ -0,0 +1,60 @@
+/* SPDX-License-Identifier: LGPL-2.1-or-later */
+
+#include "efifirmware.h"
+#include "util.h"
+#include <endian.h>
+
+static bool efifw_validate_header(const void *blob, size_t uki_efifw_len,
+                                  uint32_t *fwid_off, uint32_t *payload_off) {
+        if ((uintptr_t) blob % alignof(EfiFwHeader) != 0)
+                return false;
+        const EfiFwHeader *fw_header = ASSERT_PTR(blob);
```

Comment:
this is invalid C code. According to C standards you are not allowed to have pointers to structs of a certain type if the struct doesn't fit in the valid memory. Thus you have to check *before* you cast this here to EfiFwHeader if uki_efifw_len is actually at least as large as that structure. only then the cast is allowed.